File: medium/container-with-most-water/container-with-most-water.py
from typing import List


def max_area(height: List[int]) -> int:
    if len(height) < 2:
        return 0

    left_pointer_index = 0
    right_pointer_index = len(height) - 1
    max_area_value = 0

    while(left_pointer_index < right_pointer_index):
        min_value = height[left_pointer_index]
        if(height[right_pointer_index] < min_value):
            min_value = height[right_pointer_index]
        index_distance = right_pointer_index - left_pointer_index
        cur_area_value = min_value * index_distance
        if cur_area_value > max_area_value:
            max_area_value = cur_area_value

        if min_value is height[left_pointer_index]:
            left_pointer_index += 1
        else:
            right_pointer_index -= 1

    return max_area_value


height_data = [1, 8, 6, 2, 5, 4, 8, 3, 7]
max_area_result = max_area(height_data)

print(max_area_result)

File: medium/container-with-most-water/container-with-most-water.js
/**
 * Function that will return max area from a list of array
 *
 * @param {Array<number>} height Array to iterate
 * @returns {number}  max area results
 */

var maxArea = function(height) {
    if(height.length < 2){
        return 0
    }
    maxAreaResult = 0
    let leftPointer = 0
    let rightPointer = height.length - 1

    while(leftPointer < rightPointer){
        minHeight = Math.min(height[leftPointer], height[rightPointer])
        areaLength = rightPointer - leftPointer
        curAreaResult = minHeight * areaLength
        maxAreaResult = Math.max(curAreaResult, maxAreaResult)

        if(minHeight == height[leftPointer]){
            leftPointer += 1
        }
        else{
            rightPointer -= 1
        }
    }

    return maxAreaResult
};

heightData = [1, 8, 6, 2, 5, 4, 8, 3, 7]
result = maxArea(heightData)

console.log(result)
